pyci/rdm/constraints.py

```diff
@@ -309,6 +329,81 @@ def calc_T2(gamma, N, conjugate=False):
         a_bar - (term_6 - term_7 - term_8 + term_9)
 
 
-def calc_T2_prime():
-    pass
+def calc_T2_prime(gamma, N, conjugate=False):
+    """
+    Calculating T2' tensor
+
+    Parameters
+    ----------
+    gamma: np.ndarray
+        1DM tensor
+    N: int
+        number of electrons in the system
+    conjugate: bool
+        conjugate or regular condition
+
+    Returns
+    -------
+    np.ndarray
+
+    Notes
+    -----
+    T2' is defined as:
+
+    .. math::
+        \begin{aligned}
+        \mathcal{T}'_{2}(\Gamma)
+        = \left( \begin{matrix}
+            \mathcal{T}_{2}(\Gamma)_{\alpha \beta \gamma; \delta \epsilon \zeta} & (\Gamma_\omega)_{\alpha \beta \gamma; \nu} \\
+            (\Gamma_\omega^{\dagger})_{\mu; \delta \epsilon \zeta} & (\Gamma_\rho)_{\mu \nu}
+        \end{matrix} \right)
+        \end{aligned}
 
+        \begin{aligned}
+        \mathcal{T}'^{\dagger}_{2}(\Gamma)_{\alpha \beta; \gamma \delta} 
+        = &
+            \mathcal{T}^{\dagger}_{2}(A_{\mathcal{T}}) 
+            + (\Gamma_{\omega})_{\alpha \beta \delta; \gamma} 
+            + (\Gamma_{\omega})_{\gamma \delta \beta; \alpha} 
+        \\
+        &
+            - (\Gamma_{\omega})_{\alpha \beta \gamma; \delta} 
+            - (\Gamma_{\omega})_{\gamma \delta \alpha; \beta} 
+        \\
+        &
+            + \frac{1}{N-1} 
+            \left( 
+                \delta_{\beta \delta} (\Gamma_{\rho})_{\gamma \alpha} 
+                - \delta_{\alpha \delta} (\Gamma_{\rho})_{\gamma \beta} 
+                - \delta_{\beta \gamma} (\Gamma_{\rho})_{\delta \alpha} 
+                + \delta_{\alpha \gamma} (\Gamma_{\rho})_{\delta \beta} 
+            \right)
+    \end{aligned}
+    """
+    omega = np.einsum('abgd -> abdg', gamma)
+    rho = 1/(N-1) * np.einsum('abgb -> ag', gamma)
+
+    if not conjugate:
+        t2 = calc_T2(gamma, N, False)
+        omega_d = omega.conj().T
```

Comment:
I don't think we need to create conjugate omega in here. We can use formula 2.67 from Poelmans' thesis:
```
    omega = np.einsum('abgd -> abdg', gamma)
    rho = 1/(N-1) * np.einsum('abgb -> ag', gamma)

    if not conjugate:
        t2 = calc_T2(gamma, N, False)
        n = t2.shape[0]
        return np.block([
            [flat_tensor(t2, (n**3, n**3)), flat_tensor(omega, (n**3, n))],
            [flat_tensor(omega, (n, n**3)), rho]
        ])
```
Also, we need to import flat_tensor function from the pyci.rdm.tools module at the beginning of the file
```
from pyci.rdm.tools import flat_tensor
```